src/dependency-manager/src/manager.ts

```diff
@@ -130,14 +130,22 @@ export default abstract class DependencyManager {
    */
   protected async loadDatastores(parent_node: DependencyNode) {
     for (const [ds_name, ds_config] of Object.entries(parent_node.service_config.getDatastores())) {
+      const image_or_host: { [key: string]: string } = {};
+      const environment_service_config = this.environment.getServices()[`${parent_node.name}:${parent_node.tag}`];
+      if (environment_service_config ?.datastores[ds_name] ?.host) {
+        image_or_host['host'] = environment_service_config.datastores[ds_name].host!;
+      } else {
+        image_or_host['image'] = ds_config.docker.image;
+      }
+
       const dep_node = new DatastoreNode({
+        ...image_or_host,
         key: ds_name,
         service_config: parent_node.service_config,
-        image: ds_config.docker.image,
         tag: parent_node.tag,
         ports: {
           target: ds_config.docker.target_port,
-          expose: await this.getServicePort(),
+          expose: image_or_host['image'] ? await this.getServicePort() : ds_config.docker.target_port,
         },
```

Comment:
After reading through this, I think the idea of a host override demands a new node type like `ExternalNode` for us to parse and handle as if it is not a deployable unit. Once we know that a dependency already exists and has a host address, we don't really need any other information.